src/System.Security.Principal.Windows/src/System/Security/Principal/NTAccount.cs

```diff
@@ -370,9 +370,13 @@ private static IdentityReferenceCollection TranslateToSids(IdentityReferenceColl
             }
             finally
             {
-                LsaHandle.Dispose();
-                ReferencedDomainsPtr.Dispose();
-                SidsPtr.Dispose();
+                if (!LsaHandle.IsInvalid)
```

Comment:
Why not use the same initialize to null plan for LsaHandle as well? It would make the code look more consistent and easier to follow.

Reply:
Sure TYVM!

Reply:
also `SafeLsaReturnBufferHandle` i think

Reply:
Done!